crates/rpc/rpc/src/eth/api/fees.rs

```diff
@@ -153,8 +153,11 @@ where
             }
 
             for header in &headers {
+                let mut ratio = header.gas_used as f64 / header.gas_limit as f64;
+                if ratio.is_infinite() { ratio = 1.0 };
+
                 base_fee_per_gas.push(header.base_fee_per_gas.unwrap_or_default() as u128);
-                gas_used_ratio.push(header.gas_used as f64 / header.gas_limit as f64);
```

Comment:
prefer to check-zero-then-divide style

```rust
gas_used_ratio.push(if header.gas_limit > 0 {header.gas_used as f64 / header.gas_limit as f64} else {1.0});
```

Reply:
checking the ratio output is more straight forward here

Reply:
How? I am supposed to know that f64 divide by zero = inf?

Reply:
point taken, changed it